### PR TITLE
[homeconnect] Returns an empty list of options for an unsupported program

### LINKED ISSUE #10689: [homeconnect] Avoid repeated requests when the request failed
Here are the first messages exchanged with my washer:

image

As you can see, the current selected mode is not a mode supported by the API. The request is returning code 404.

Maybe it could be improved to avoid requesting it several times (3 times) when it first failed.

### PR DESCRIPTION
**Pull request by lolodomo**
Fix #10689

Signed-off-by: Laurent Garnier <lg.hc@free.fr>

### CODE REVIEW

**Comment by lolodomo** [comment]
@bruestel : for your review (I don't understand why you are not automatically assigned as reviewer as the binding author - I cannot find you in the reviewer list).

**Comment by bruestel** [comment]
> @bruestel : for your review (I don't understand why you are not automatically assigned as reviewer while you are the binding author - I cannot find you in the reviewer list).

Strange, I will have a look tomorrow. I'm on the move today. Thanks for your MR.

File: bundles/org.openhab.binding.homeconnect/src/main/java/org/openhab/binding/homeconnect/internal/client/HomeConnectApiClient.java [hunk 1]
```diff
@@ -633,12 +630,21 @@ public void stopProgram(String haId)
         Request request = createRequest(HttpMethod.GET, BASE_PATH + haId + "/programs/available/" + programKey);
         try {
             ContentResponse response = sendRequest(request, apiBridgeConfiguration.getClientId());
-            checkResponseCode(HttpStatus.OK_200, request, response, haId, null);
+            checkResponseCode(asList(HttpStatus.OK_200, HttpStatus.NOT_FOUND_404), request, response, haId, null);
```
Comment (t-8ch) [inline bundles/org.openhab.binding.homeconnect/src/main/java/org/openhab/binding/homeconnect/internal/client/HomeConnectApiClient.java:633]: `List.of()` would be more Java-11-y

File: bundles/org.openhab.binding.homeconnect/src/main/java/org/openhab/binding/homeconnect/internal/client/HomeConnectApiClient.java [hunk 2]
```diff
@@ -633,12 +630,21 @@ public void stopProgram(String haId)
...
+            // Code 404 accepted only if the returned error is "SDK.Error.UnsupportedProgram"
+            if (response.getStatus() == HttpStatus.NOT_FOUND_404
+                    && (responseBody == null || !responseBody.contains("SDK.Error.UnsupportedProgram"))) {
+                throw new CommunicationException(HttpStatus.NOT_FOUND_404, response.getReason(),
+                        responseBody == null ? "" : responseBody);
+            }
+
+            List<AvailableProgramOption> availableProgramOptions = response.getStatus() == HttpStatus.OK_200
+                    ? mapToAvailableProgramOption(responseBody, haId)
+                    : new ArrayList<>();
```
Comment (t-8ch) [inline bundles/org.openhab.binding.homeconnect/src/main/java/org/openhab/binding/homeconnect/internal/client/HomeConnectApiClient.java:647]: Also `List.of()`.

**Review by bruestel — APPROVED** [review]
Looks good to me. Thank you.

**Comment by lolodomo** [comment]
@fwolter : can you merge, please.

**Review by fwolter — APPROVED** [review]

**Comment by lolodomo** [comment]
Thank you

**Pull request merged**